lighthouse-core/audits/large-javascript-libraries.js

```diff
@@ -0,0 +1,165 @@
+  title: 'Avoids unnecessarily large JavaScript libraries',
+  /** Title of a Lighthouse audit that provides detail on large Javascript libraries that are used on the page that have better alternatives. This descriptive title is shown when to users when some known unnecessarily large libraries are detected on the page.*/
+  failureTitle: 'Replace unnecessarily large JavaScript libraries',
+  /** Description of a Lighthouse audit that tells the user why they should care about the large Javascript libraries that have better alternatives. This is displayed after a user expands the section to see more. No character length limits. */
+  description: 'Large JavaScript libraries can lead to poor performance. ' +
+    'Prefer smaller, functionally equivalent libraries to reduce your bundle size.' +
+    ' [Learn more](https://developers.google.com/web/fundamentals/performance/webpack/decrease-frontend-size#optimize_dependencies).',
+  /** Label for a column in a data table. Entries will be names of large JavaScript libraries that could be replaced. */
+  columnLibraryName: 'Library',
+  /** [ICU Syntax] Label for the Large JavaScrip Libraries audit identifying how many large libraries were found. */
+  displayValue: `{libraryCount, plural,
+    =1 {1 large library found}
+    other {# large libraries found}
+    }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LargeJavascriptLibraries extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'large-javascript-libraries',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['Stacks'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {Array<{original: MinifiedBundlePhobiaLibrary, suggestions: MinifiedBundlePhobiaLibrary[]}>} */
+    const libraryPairings = [];
+    const detectedLibs = artifacts.Stacks.filter(stack => stack.detector === 'js');
+
+    const seenLibraries = new Set();
+
+    for (const detectedLib of detectedLibs) {
+      if (!detectedLib.npm || !libStats[detectedLib.npm]) continue;
+
+      const suggestions = librarySuggestions[detectedLib.npm];
+      if (!suggestions) continue;
+
+      if (seenLibraries.has(detectedLib.npm)) continue;
+      seenLibraries.add(detectedLib.npm);
+
+      let version = 'latest';
+      if (detectedLib.version && libStats[detectedLib.npm].versions[detectedLib.version]) {
+        version = detectedLib.version;
+      }
+
+      const originalLib = libStats[detectedLib.npm].versions[version];
+
+      /** @type {Array<{name: string, repository: string, gzip: number}>} */
+      const smallerSuggestions = [];
+      for (const suggestion of suggestions) {
+        if (libStats[suggestion].versions['latest'].gzip > originalLib.gzip) continue;
```

Comment:
```suggestion
        if (libStats[suggestion].versions['latest'].gzip >= originalLib.gzip) continue;
```

Reply:
was there discussion about only flagging if the difference is x or more bytes difference?

Reply:
yeah, patrick proposed we don't store anything within 512 bytes of the latest version size, since rounding means it likely won't affect what the user sees, and also a smaller database will reduce the overall LH bundle size

Reply:
I think those are slightly separate discussions @saavannanavati.

I'm not sure we strictly need this `continue` because we should never add a suggestion to our set if it's bigger than the target library :) but it feels reasonable to leave in as a baseline check